libsolidity/ast/Types.h

```diff
@@ -338,9 +384,15 @@ class IntegerConstantType: public Type
 
 	/// @returns the smallest integer type that can hold the value or an empty pointer if not possible.
 	std::shared_ptr<IntegerType const> integerType() const;
+	/// @returns the smallest fixed type that can hold the value or an empty pointer
```

Comment:
Should be changed to "... hold the value or incurs the least precision loss. If the integer part does not fit, returns an empty pointer."